src/pyhf/optimize/opt_scipy.py

```diff
@@ -28,26 +30,50 @@ def minimize(
 
         Returns:
             bestfit parameters
-        
+
         """
+        tensorlib, _ = get_backend()
+
+        all_idx = default_backend.astensor(range(pdf.config.npars), dtype='int')
+        all_init = default_backend.astensor(init_pars)
+
         fixed_vals = fixed_vals or []
-        indices = [i for i, _ in fixed_vals]
-        values = [v for _, v in fixed_vals]
-        constraints = [{'type': 'eq', 'fun': lambda v: v[indices] - values}]
+        fixed_values = [x[1] for x in fixed_vals]
+        fixed_idx = [x[0] for x in fixed_vals]
+
+        variable_idx = [x for x in all_idx if x not in fixed_idx]
+        variable_init = all_init[variable_idx]
+        variable_bounds = [par_bounds[i] for i in variable_idx]
+
+        tv = _TensorViewer([fixed_idx, variable_idx])
+
+        data = tensorlib.astensor(data)
+        fixed_values_tensor = tensorlib.astensor(fixed_values, dtype='float')
+
+        def func(pars):
+            pars = tensorlib.astensor(pars)
+            constrained_pars = tv.stitch([fixed_values_tensor, pars])
+            return objective(constrained_pars, data, pdf)
+
         result = minimize(
-            objective,
-            init_pars,
-            constraints=constraints,
+            func,
```

Comment:
Maybe
```suggestion
            _objective,
```
to keep it clear?

Reply:
func is in all the other backends right now, so we're going for consistency. We can rename in the next PR that completely removes all of these changes.